lib/vtls/gtls.c

```diff
@@ -845,10 +845,13 @@ static CURLcode gtls_client_init(struct Curl_cfilter *cf,
   init_flags |= GNUTLS_FORCE_CLIENT_CERT;
 #endif
 
-#if defined(GNUTLS_NO_TICKETS)
-  /* Disable TLS session tickets */
-  init_flags |= GNUTLS_NO_TICKETS;
-#endif
+  /* Disable TLS session tickets for tlsv1.2
+   * Its incredibly strange, enabling it makes gnutls report and store zero
+   * length session tickets on tls1.2 connections
+   * */
+  if(config->version == CURL_SSLVERSION_TLSv1_2) {
+    init_flags |= GNUTLS_NO_TICKETS;
+  }
```

Comment:
This change removes the `#if defined(GNUTLS_NO_TICKETS)`. Why? This breaks builds with older GnuTLS versions.

Reply:
Apologies, that was not intended